src/components/PokeList.js

```diff
@@ -192,10 +205,13 @@ const PokeList = ({
             abilityOptions={abilityOptions}
             setAbilityOptions={setAbilityOptions}
             setSortType={setSortTypeFlags}
+            setFilterType={setFilterType}
           />
           <div className="card-container">
-          {/* By default, display all Pokemons, else show  filtered/sorted cards */}
-            {generateSortedCards(userDidSort ? filteredPokemons : allPokemons)}
+            {/* By default, display all Pokemons, else show  filtered/sorted cards */}
+            {generateSortedCards(
+              userDidSort || userDidFilter ? filteredPokemons : allPokemons
```

Comment:
Is it necessary to generate sorted cards of allPokemons? I notice above that you set allPokemonsFiltered to be allPokemons and then set filteredPokemons to allPokemonsFiltered. Wouldn't allPokemons = filteredPokemons at first and then when filter the data, allPokemonsFiltered gets updated? It is hard to tell without seeing all the code. Disregard this if it is totally wrong.

Reply:
Good question.  Also, I'm not sure there's a need to differentiate between whether the user sorted or filtered.  Either way, it derives `allPokemonsFiltered`.